pkg/controller/llmisvc/DEV-authz.md

```diff
@@ -0,0 +1,205 @@
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: rhcl-operator
+  namespace: kuadrant-system
+spec:
+  channel: stable
+  installPlanApproval: Automatic
+  name: rhcl-operator
+  source: redhat-operators
+  sourceNamespace: openshift-marketplace
+---
+kind: OperatorGroup
+apiVersion: operators.coreos.com/v1
+metadata:
+  name: kuadrant
+  namespace: kuadrant-system
+spec:
+  upgradeStrategy: Default
+EOF
+
+while [ $(kubectl get pod -n kuadrant-system  | wc -l) -le 1 ]; 
+do
+  echo "⏳ waiting for Kuadrant Pod to appear…"
+  sleep 10
+done
+kubectl wait --for=condition=ready pod -l app=kuadrant -n kuadrant-system --timeout 150s
+
+kubectl apply -f - <<EOF
+apiVersion: kuadrant.io/v1beta1
+kind: Kuadrant
+metadata:
+  name: kuadrant
+  namespace: kuadrant-system
+EOF
+```
+
+**Create Pull Secret**
+
+Refer to [this doc](https://docs.redhat.com/en/documentation/red_hat_connectivity_link/1.0/html-single/installing_connectivity_link_on_openshift/index#auth-registry-wasm-plugin)
+
+You can get the secret from [this site](https://access.redhat.com/terms-based-registry/) and create it in `api-gateway` namespace and gateway namespace(`openshift-ingress`)
+
+```shell
+kubectl create ns api-gateway
+
+kubectl create secret docker-registry wasm-plugin-pull-secret -n api-gateway \ --docker-server=registry.redhat.io \ --docker-username=<your-registry-service-account-username> \ --docker-password=<your-registry-service-account-password>
+
+kubectl create secret docker-registry wasm-plugin-pull-secret -n openshift-ingress \ --docker-server=registry.redhat.io \ --docker-username=<your-registry-service-account-username >\ --docker-password=<your-registry-service-account-password>
+
+```
+
+**Update gateway label**
+```shell
+kubectl label gateway openshift-ai-inference -n openshift-ingress kuadrant.io/gateway="true"
+```
+
+**Authentication with KubernetesTokenReview**
+
+```shell
+kubectl apply -f - <<EOF
+apiVersion: kuadrant.io/v1
+kind: AuthPolicy
+metadata:
+  name: llm-test-authn
+  namespace: $TEST_NS
+spec:
+  targetRef:
+    group: gateway.networking.k8s.io
+    kind: HTTPRoute
+    name: $LLM_ISVC_NAME-kserve-route
+  defaults:
+    strategy: merge
+    rules:
+      authentication:
+        kubernetes-user:
+          credentials:
+            authorizationHeader: {}
+          kubernetesTokenReview:
+            audiences:
+            - https://kubernetes.default.svc
+          metrics: false
+          priority: 0
+      authorization:
+        kubernetes-rbac:
+          kubernetesSubjectAccessReview:
+            resourceAttributes:
+              group:
+                value: gateway.networking.k8s.io/v1
+              name:
+                value: $LLM_ISVC_NAME-kserve-route
+              namespace:
+                value: $TEST_NS
+              resource:
+                value: httproutes
+              subresource:
+                value: ""
+              verb:
+                value: get
```

Comment:
why HTTPRoute?

Reply:
lol :) it is just sample. for isvc, we used `get` `isvc` So we might change it to llmisvc : https://github.com/opendatahub-io/odh-model-controller/blob/incubating/internal/controller/resources/template/authconfig_isvc_userdefined.yaml